### PR TITLE
Supermatter delamination tweaks

### PR DESCRIPTION
**Pull request by SapphicOverload**
# Document the changes in your pull request

Supermatter explosion radius and EMP radius is now based on the energy of the crystal instead of the power of the gas mixture inside the chamber, and will explode harder if it's in a vacuum. This is to disincentivize using an RCD on the floor as an easy fix to buy time with no consequences. The delamination will now also irradiate what's left of the engine, so repair and cleanup will take a bit more effort than just patching the breach.

# Changelog

:cl:  
tweak: supermatter explosion and EMP radius now based on the energy of crystal instead of the gases in the chamber
tweak: supermatter explodes harder in a vacuum
tweak: supermatter will now irradiate what's left of engineering when it explodes
/:cl:

### CODE REVIEW

**Comment by ToasterBiome** [comment]
I liked the fact that exploding the crystal that destroys anything it touches has actual effects on the station instead of “i guess it’s just gone? patch the hole in 3 seconds”

**Comment by iamthesilly** [comment]
> I liked the fact that exploding the crystal that destroys anything it touches has actual effects on the station instead of “i guess it’s just gone? patch the hole in 3 seconds”

well you're in luck then, i made it irradiate what's left of the engine to make repair and cleanup take a bit more than just patching the breach

**Comment by JamieD1** [comment]
Pictures of a couple different explosions?

**Comment by iamthesilly** [comment]
> Pictures of a couple different explosions?

explosion at 0 EER (unpowered)
image

explosion at 2700 EER
image

explosion at 6000 EER
image
image

**Comment by MegaEmpirical** [comment]
Now buff the SM event too.

**Review by ThatLing — APPROVED** [review]
What's the worst that could happen

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -72,6 +72,7 @@
 //These would be what you would get at point blank, decreases with distance
 #define DETONATION_RADS 200
 #define DETONATION_HALLUCINATION 600
+#define SUPERMATTER_EXPLOSION_LAMBDA 20
```
Comment (ThatLing) [inline code/modules/power/supermatter/supermatter.dm:75]: #undef this at the bottom of the file

**Comment by MajManatee** [comment]
@SapphicOverload add the undef and ill merge it

**Comment by iamthesilly** [comment]
> @SapphicOverload add the undef and ill merge it

done

**Comment by ShoziOffical** [comment]
oh no now engineers gonna be terrified

**Pull request merged**